### PR TITLE
chore(redux-thunk): fix typings

### PR DESCRIPTION
**Pull request by 121watts**
Prerequisite for normalization.  We had no types for `redux-thunk` and / or they were typed incorrectly.

### CODE REVIEW

File: ui/src/buckets/actions/index.ts [hunk 1]
```diff
@@ -180,6 +185,9 @@ export const deleteBucket = (id: string, name: string) => async (
 
     dispatch(removeBucket(id))
     dispatch(checkBucketLimits())
+    dispatch(() => ({
+      type: 'PLERPS',
```
Comment (ebb-tide) [inline ui/src/buckets/actions/index.ts]: ? :D

File: ui/src/labels/actions/index.ts [hunk 2]
```diff
@@ -17,7 +17,12 @@ import {
 } from 'src/shared/copy/notifications'
 import {GetState} from 'src/types'
 
-export type Action = SetLabels | AddLabel | EditLabel | RemoveLabel
+export type Action =
+  | SetLabels
+  | AddLabel
+  | EditLabel
+  | RemoveLabel
+  | NotifyAction
```
Comment (ebb-tide) [inline ui/src/labels/actions/index.ts]: Aren't we using this Action to define what the reducer can accept? I'm afraid this reduces the type safety of the reducer. What would you think about removing line 25 and adding another type `ThunkActions = Action | NotifyAction` that we could use for dispatch?

File: ui/src/timeMachine/actions/index.ts [hunk 3]
```diff
@@ -100,6 +100,11 @@ export type Action =
   | ReturnType<typeof updateCheckThreshold>
   | ReturnType<typeof removeCheckThreshold>
 
+type ExternalActions =
```
Comment (ebb-tide) [inline ui/src/timeMachine/actions/index.ts]: ooh love this! Maybe we could even import and export all Actions from a central place?

File: ui/src/buckets/actions/index.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import {Dispatch} from 'redux-thunk'
+import {Dispatch} from 'react'
```
Comment (hoorayimhelping) [inline ui/src/buckets/actions/index.ts]: can you give some background in this change _por favor_? like, why is it being imported from React now?

File: ui/src/timeMachine/actions/index.ts [hunk 5]
```diff
@@ -100,6 +100,11 @@ export type Action =
...
+  | ReturnType<typeof loadBuckets>
+  | ReturnType<typeof saveAndExecuteQueries>
+  | ReturnType<typeof setValues>
```
Comment (hoorayimhelping) [inline ui/src/timeMachine/actions/index.ts]: do all the the functions that rely on external actions call all these functions, or is this just the set of all the external functions?

File: ui/src/labels/actions/index.ts (hunk 2 above)
Comment (121watts) [reply ui/src/labels/actions/index.ts]: What I ultimately want to do is centralize all actions into a single format and location.  So, yes!

File: ui/src/timeMachine/actions/index.ts (hunk 3 above)
Comment (121watts) [reply ui/src/timeMachine/actions/index.ts]: Yes, I want to do that in another PR.

File: ui/src/timeMachine/actions/index.ts (hunk 5 above)
Comment (121watts) [reply ui/src/timeMachine/actions/index.ts]: "`ExternalActions`" was just a name I thought of for actions not declared in this file.  Really, I want a central repository and format for all action types.

File: ui/src/buckets/actions/index.ts (hunk 4 above)
Comment (121watts) [reply ui/src/buckets/actions/index.ts]: `redux-thunk` doesn't actually have an exported`Dispatch` type.  So, `Dispatch` was really `any`

File: ui/src/buckets/actions/index.ts (hunk 1 above)
Comment (121watts) [reply ui/src/buckets/actions/index.ts]: oops, ill pull that out.  thanks for spotting that.

**Pull request merged**